src/ProvidedTypes.fs

```diff
@@ -14365,8 +14368,11 @@ namespace ProviderImplementation.ProvidedTypes
 
                 match methodBaseT with
                 | :? ProvidedMethod as mT when (match methodBaseT.DeclaringType with :? ProvidedTypeDefinition as pt -> pt.IsErased | _ -> true) ->
-                    let exprT = mT.GetInvokeCode(Array.toList parametersT)
-                    check exprT
+                    match mT.GetInvokeCode with
+                    | Some invokeCode ->
+                        let exprT = invokeCode(Array.toList parametersT)
+                        check exprT
+                    | None -> <@@ () @@>

```

Comment:
I wasn't sure what to do here. @dmitry-a-morozov suggested `Unchecked.defaultof<_>`, but generative providers appear to fail due to a null reference. @7sharp9 suggested this solution, so I've made a go with it as a first pass.

@dmitry-a-morozov, would you be able to add some test cases?

Reply:
@panesofglass I'm sorry but no. I don't have capacity to work on it now.